buildtest/system.py

```diff
@@ -23,20 +24,30 @@ def __init__(self):
         """Constructor method for BuildTestSystem(). Defines all system configuration using
            class variable **system** which is a dictionary
         """
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Starting System Compatibility Check")
         self.init_system()
-        self.system["SYSTEM"] = platform.system()
+        self.system["platform"] = platform.system()
         self.scheduler = self.check_scheduler()
         self.check_lmod()
 
+        if self.system["platform"] != "Linux":
+            print("System must be Linux")
+            sys.exit(1)
+
+        self.logger.debug("Finished System Compatibility Check")
+
     def init_system(self):
         """Based on the module "distro" set the linux distrubution name and version
         """
-        self.system["OS_NAME"] = distro.linux_distribution(
+        self.system["os_name"] = distro.linux_distribution(
             full_distribution_name=False
         )[0]
-        self.system["OS_VERSION"] = distro.linux_distribution(
-            full_distribution_name=False
-        )[1]
+        self.system["os_ver"] = distro.linux_distribution(full_distribution_name=False)[
```

Comment:
What is the namespace for these keys? E.g., why are we using "os_ver" which is sort of short /. choppy?

Reply:
good point right now these are only used for logging, there was some additional work in this PR https://github.com/HPC-buildtest/buildtest-framework/pull/236/files#diff-da5c353202a52de72277768aebc926b6R35-R44 which i kept open but won't get merged. Idea was to use BuildTestSystem to detect system detail some of which is auto-detected.

As per ``os_ver`` and ``os_name`` i was trying to do a few things
1. Standardize all keys to be lower case and multi-words separated via underscore _
2. Capture OS details as part of logging, could be used for logging or reporting as part of run results.
3. Before buildtest was doing sanity check of binaries by system packages. Initially it supported rpm so i had a requirement for buildtest to only support Centos/Fedora/RHEL based system. That feature is dropped so there is no real purpose of having os_name and os_ver other than logging.

Reply:
Is there any reason to truncate the name? I like `os_name`, but would it make sense to do `os_version` so it's explicitly clear?

Reply:
ill make the change to ``os_version`` makes sense. Then no truncation for time being. If you see in the other PR https://github.com/HPC-buildtest/buildtest-framework/pull/236/files#diff-da5c353202a52de72277768aebc926b6R39-R41 i can go a bit more elegant such that we can 
reference something like 

```
self.system['os']['name']
self.system['os']['version']
self.system['os']['major']
self.system['os']['major']
self.system['os']['build_number']
```

Again we can go crazy with this and would be fine with making any change that makes sense. Let me know your thoughts or a proposal

Reply:
What you showed above is lovely! No worries about adding it here, as you mentioned it can be a separate PR.

Reply:
Okay i made some changes in last commit just now to add a few extra details. Here is a snipet of the log, i think adding the environment section can be useful in troubleshooting.


```
(buildtest-framework) ssi29@ag-mxg-hulk090> head -n 40 buildtest.log
2020-04-07 15:04:54,124 [main.py:19 -  main() ] - [INFO] Starting buildtest log
2020-04-07 15:04:54,124 [system.py:28 - __init__() ] - [DEBUG] Starting System Compatibility Check
2020-04-07 15:04:54,196 [system.py:58 - init_system() ] - [INFO] Operating System: rhel
2020-04-07 15:04:54,197 [system.py:59 - init_system() ] - [INFO] Operating System Version: 7.6
2020-04-07 15:04:54,197 [system.py:60 - init_system() ] - [INFO] Operating System Major Version: 7
2020-04-07 15:04:54,197 [system.py:61 - init_system() ] - [INFO] Operating System Minor Version: 7
2020-04-07 15:04:54,197 [system.py:62 - init_system() ] - [INFO] Operating System Build Number:
2020-04-07 15:04:54,197 [system.py:64 - init_system() ] - [INFO] Session Environment Variables
2020-04-07 15:04:54,197 [system.py:65 - init_system() ] - [INFO] ________________________________________________________________________________
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] REMOTEHOST: lt4-th3-dsa.dsone.3ds.com
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] CABAQUS: /usr/abaqus60/udev/source
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] EBVERSIONZLIB: 1.2.11
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] MANPATH: /mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/man:/mxg-hpc/users/ssi29/easybuild/software/libffi/3.2.1-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/XZ/5.2.4-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/SQLite/3.29.0-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/Tcl/8.6.9-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/Tcl/8.6.9-GCCcore-8.3.0/man:/mxg-hpc/users/ssi29/easybuild/software/libreadline/8.0-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/ncurses/6.1-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/zlib/1.2.11-GCCcore-8.3.0/share/man:/mxg-hpc/users/ssi29/easybuild/software/bzip2/1.0.8-GCCcore-8.3.0/man:/mxg-hpc/users/ssi29/easybuild/software/GCCcore/8.3.0/share/man:/r/cluster_software/pub_lsf/10.1/man:/usr/local/man:/usr/share/man:/usr/man:/usr/X11R6/man:/usr/openwin/man:/usr/share/man/allman:/usr/abaqus60/udev/generated/man:/usr/abaqus60/udev/object/lnx86_64/External/graphviz/man
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] FIGS: /p/marketing/figures
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] EBVERSIONPYTHON: 3.7.4
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] XDG_SESSION_ID: 144609
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] CABAQUS_O: /usr/abaqus60/udev/object/lnx86_64
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] HOSTNAME: ag-mxg-hulk090
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] EBDEVELPYTHON: /mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/easybuild/Python-3.7.4-GCCcore-8.3.0-easybuild-devel
2020-04-07 15:04:54,197 [system.py:67 - init_system() ] - [INFO] _ModuleTable003_: Zy1ocGMvdXNlcnMvc3NpMjkvZWFzeWJ1aWxkL21vZHVsZXMvYWxsL1NRTGl0ZS8zLjI5LjAtR0NDY29yZS04LjMuMC5sdWEiLFsiZnVsbE5hbWUiXT0iU1FMaXRlLzMuMjkuMC1HQ0Njb3JlLTguMy4wIixbImxvYWRPcmRlciJdPTcscHJvcFQ9e30sWyJzdGFja0RlcHRoIl09MSxbInN0YXR1cyJdPSJhY3RpdmUiLFsidXNlck5hbWUiXT0iU1FMaXRlLzMuMjkuMC1HQ0Njb3JlLTguMy4wIix9LFRjbD17WyJmbiJdPSIvbXhnLWhwYy91c2Vycy9zc2kyOS9lYXN5YnVpbGQvbW9kdWxlcy9hbGwvVGNsLzguNi45LUdDQ2NvcmUtOC4zLjAubHVhIixbImZ1bGxOYW1lIl09IlRjbC84LjYuOS1HQ0Njb3JlLTguMy4wIixbImxvYWRPcmRlciJdPTYscHJvcFQ9e30sWyJzdGFja0RlcHRo

```

I think in future it would help capture paths to launchers (srun, sbatch, mpirun, mpiexec) I had some ideas in buildtest setting to allow user to specify path to launcher paths let's say in case of mpirun you or other mpi specific launcher you dont want to rely what's in $PATH as it can change due to shell or changes to module so to ensure consistency it would be useful to let user specify a full path to launcher. Again out of scope for this task.  

For now i think it's safe to say this issue is resolved. I changed ``[os_ver]`` to ``['os']['version']``

Reply:
Great! One final tweak - all of those variables should be for level DEBUG and not INFO.

Reply:
makes sense i will make that change, should i make all of ``self.system`` be DEBUG to be consistent?

Reply:
ill make it DEBUG for all them to be consistent.

Reply:
yes.